prov/sockets/src/sock_cntr.c

```diff
@@ -301,6 +304,10 @@ static int sock_cntr_wait(struct fid_cntr *cntr, uint64_t threshold,
 		ret = fi_wait_cond(&_cntr->cond, &_cntr->mut, timeout);
 	}
 
+	if (atomic_get(&_cntr->value) >= threshold &&
+	    atomic_get(&_cntr->last_read_val) < threshold) {
```

Comment:
> Yeah, it is possible (eg: timeout occured, and the current value is less than threshold).

Oops, I think I asked the wrong question. What I meant to ask was whether `&_cntr->last_read_val` would ever be greater than or equal to `threshold`. 

I'm wondering whether you could just do:

```c
if (atomic_get(&_cntr->value) >= threshold)
    atomic_set(&_cntr->last_read_val, threshold);
```

Also, should this be setting `&_cntr->last_read_val` to `threshold` or `&_cntr->value`?